File: Project Euler #30/Java/Solution30.java
import java.util.Scanner;

public class Solution30 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int sum = 0;
        for(int i=100;i<1000000;i++) {
            int num = i;
            int temp = 0;
            while(num>0) {
                temp += (int)Math.pow(num%10, n);
                num /=10;
            }
            if(temp == i) {
                sum+=i;
            }
        }
        System.out.println(sum);
    }
}

File: Project Euler #30/C/solution30.c
#include <stdio.h>
#include <math.h>

int main() {

    int n;
    scanf("%d", &n);
    int sum=0, temp=0;
    for(int i=100;i<1000000;i++){
        int num = i;
        temp = 0;
        while(num){
            temp += pow(num%10, n);
            num/=10;
        }
        if(temp==i){
            sum+=i;
        }
    }
    printf("%d", sum); 
    return 0;
}
